### DISCUSSION TITLE
The oal script is customized for alarms, and some endpoints will not trigger alarms

### DISCUSSION THREAD

**Discussion by Mxgxg** [apache/skywalking#7775] [Q&A] [answered]
I customized an oal script to count the error rate of the endpoint and trigger an alarm.
In the tracking, it has been marked as an error, but in the dashboard interface, the endpoint success rate is still 100%
![390bd045e7b96524427ddc77941b2ce](https://user-images.githubusercontent.com/79822553/134451547-42f2351b-7cda-4813-866b-612fea5ee88d.png)
![95e40652cad460c1b263ab938a158b7](https://user-images.githubusercontent.com/79822553/134451584-aaa90650-d4d2-4208-801d-65f44ee69e2d.png)

![08270129e4dfeffb1f3ea3556373d6b](https://user-images.githubusercontent.com/79822553/134451595-02325e23-6d5a-4490-94e5-41762a32ecb1.png)
Because there is custom exception handling in the project, manual API tracing is used
![1d54a3e1f12d930ff82c3d20315bc8f](https://user-images.githubusercontent.com/79822553/134452271-a3ce73bf-1ed4-4aab-9a46-5347833b6167.png)

The entire trace is not marked as an error

![12958040000437dd03d78e28bc97a76](https://user-images.githubusercontent.com/79822553/134452585-bdfe82be-8eeb-4bb0-94d3-e93eebaa637f.png)

The script will match other endpoints with exceptions and trigger an alarm, which puzzles me

**Comment by wu-sheng** [comment, marked as answer]
I think you mixed 2 concepts. `Span's error status` V.S. `Endpoint's error status`. The OAL is endpoint based, the trace page only shows you this specific span status is incorrect.

**Reply by Mxgxg** [reply]
What should I do to make a span error and make him think it is an endpoint error.
In the project, a similar span error goes into custom exception handling, which I mark as an error and show as an endpoint error.
so i think this is a right way

**Reply by wu-sheng** [reply]
You need to make sure it throws to Controller level, or tag this in your controller. Basically, we need to tag the entry span status as error.

**Reply by Mxgxg** [reply]
reading exception log, not thrown to the controller level. Is fastjson throws an exception during data conversion.
Is there no way to monitor it in this case?

**Reply by wu-sheng** [reply]
I don't know what do you mean `no way`. I think I have shown you the way. You `catch` exception, which stops the stack propagation.

**Reply by Mxgxg** [reply]
Ok, I see what you mean. Thank you for you answer!

**Comment by yuyuyuzz** [comment]
> You need to make sure it throws to Controller level, or tag this in your controller. Basically, we need to tag the entry span status as error.

A single Span's error status doesn't necessarily lead to Endpoint's error status, but Span's error does mean that there is a failure in the internal logic process, which also creates some impact, if I want to monitor a single Span's error status to create an alert, is it currently possible?

**Reply by wu-sheng** [reply]
No, you can't. Alerting is request-oriented, a span's status means nothing. Some applications use exceptions to control the business logic, and some exceptions could be handled properly without performance impact.
